hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/TimelineService.java

```diff
@@ -98,16 +98,42 @@ public TimelineService(Config config) throws IOException {
     public Boolean help = false;
   }
 
+  private int startServiceOnPort(int port) throws IOException {
+    if (!(port == 0 || (1024 <= port && port < 65536))) {
+      throw new IllegalArgumentException(String.format("startPort should be between 1024 and 65535 (inclusive), "
+          + "or 0 for a random free port. but now is %s.", port));
+    }
+    int maxRetries = 16;
```

Comment:
pull into `static final` variable?

Reply:
you also probably want to add a small wait between attempts?

Reply:
> pull into `static final` variable?

done

Reply:
> you also probably want to add a small wait between attempts?

i think not need to add a small wait between attempts. because in most cases will retry when port is already in use, a attempt will change to new port .